ofrak_core/ofrak/core/esp/app.py

```diff
@@ -0,0 +1,959 @@
+            memory_offset=memory_offset,
+            segment_size=segment_size
+        )
+
+
+class ESPAppSectionAnalyzer(Analyzer[None, ESPAppSection]):
+    """
+    Analyze an ESPAppSection to extract its attributes.
+    """
+    targets = (ESPAppSection,)
+    outputs = (ESPAppSection,)
+
+    async def analyze(self, resource: Resource, config=None) -> ESPAppSection:
+        # First check if we already have the section attributes
+        try:
+            section_attrs = resource.get_attributes(ESPAppSection)  # type: ignore
+            return section_attrs
+        except:
+            pass
+            
+        # Otherwise get from structure
+        section_structure = await resource.view_as(ESPAppSectionStructure)
+        
+        # Find the corresponding header
+        parent = await resource.get_parent()
+        header = await parent.get_only_child_as_view(
+            ESPAppSectionHeader,
+            ResourceFilter(
+                tags=(ESPAppSectionHeader,),
+                attribute_filters=[
+                    ResourceAttributeValueFilter(
+                        ESPAppSectionStructure.SectionIndex, section_structure.section_index
+                    )
+                ],
+            ),
+        )
+        
+        return ESPAppSection(
+            section_index=section_structure.section_index,
+            name=header.name,
+            virtual_address=header.memory_offset,
+            size=header.segment_size
+        )
+
+
+class ESPAppAnalyzer(Analyzer[None, ESPAppAttributes]):
+    """
+    Analyze attributes of ESP app.
+    """
+
+    targets = (ESPApp,)
+    outputs = (ESPAppAttributes,)
+
+    async def analyze(self, resource: Resource, config: Optional[ESPAppConfig]) -> ESPAppAttributes:
+        config = await get_esp_app(resource, config)
+        try:
+            calculated_checksum = int(config.image.calculate_checksum())
+            checksum_valid = True if config.image.checksum == calculated_checksum else False
+            hash_valid = False
+            calculated_hash = 0
+            # append_digest is only available on ESP32 images, not ESP8266
+            if hasattr(config.image, 'append_digest') and config.image.append_digest:
+                calculated_hash = int.from_bytes(config.image.calc_digest[:4], 'little')  # Use first 4 bytes as int
+                if config.image.stored_digest == config.image.calc_digest:
+                    hash_valid = True
+                else:
+                    hash_valid = False
+
+            return ESPAppAttributes(
+                chip_name=self.__get_chip_name(config.image),
+                checksum_valid=checksum_valid,
+                calculated_checksum=calculated_checksum,
+                hash_valid=hash_valid,
+                calculated_hash=calculated_hash,
+            )
+        finally:
+            if hasattr(config, "f"):
+                config.f.close()
+
+    def __get_chip_name(self, image):
```

Comment:
```suggestion
    def _get_chip_name(self, image):
```

This and other internal methods should use single underscores